pandas/core/indexes/timedeltas.py

```diff
@@ -64,6 +64,10 @@ class TimedeltaIndex(DatetimeTimedeltaMixin):
         Optional timedelta-like data to construct index with.
     unit : {'D', 'h', 'm', 's', 'ms', 'us', 'ns'}, optional
         The unit of ``data``.
+
+        .. deprecated:: 2.2.0
+         Use ``pd.to_timedelta`` instead.
```

Comment:
Does that mean that integer inputs are always interpreted as nanos or based on the unit in `dtype`? If so, would be good to explain that here

Reply:
thats a bit ambiguous bc of #52457, #48312

Reply:
deal-breaker?

Reply:
I guess since the behavior of int inputs with `unit=None` doesn't change (albeit ambiguous) I suppose this is OK